ush/templates/FV3LAM_wflow.xml

```diff
@@ -105,12 +105,18 @@ tasks; and the "FCST" type is used for the RUN_FCST_TN task.
 
 ]>
 
-<workflow realtime="T" scheduler="&SCHED;" cyclethrottle="3" cyclelifespan="01:00:00:00">
+<workflow realtime="T" scheduler="&SCHED;" cyclethrottle="25" cyclelifespan="01:00:00:00">
 {# Double quotes are required inside the strftime! Expect an error from reading the template if using single quotes. #}
+{%- if (run_task_make_grid or run_task_make_orog or run_task_make_sfc_climo) %}
   <cycledef group="at_start">{{ cdate_first_cycl.strftime("%M %H %d %m %Y *") }}</cycledef>
-  <cycledef group="forecast">{{ cdate_first_cycl.strftime("%Y%m%d%H%M") }} {{ cdate_last_cycl.strftime("%Y%m%d%H%M") }} {{ cycl_freq }}</cycledef>
+{% endif %}
+{% for cycl_hr in cycl_hrs %}
+  <cycledef group="{{cycl_hr}}Z">{{ cdate_first_cycl.strftime("%Y%m%d") }}{{ cycl_hr }}00 {{ cdate_last_cycl.strftime("%Y%m%d") }}{{ cycl_hr }}00 24:00:00</cycledef>
+{%- endfor %}
 {%- if do_dacycle %}
-  <cycledef group="dacycle">00 03,06,09,15,18,21 {{cdate_first_cycl.strftime("%d") }}-{{ cdate_last_cycl.strftime("%d") }} {{ cdate_first_cycl.strftime("%m") }} {{cdate_first_cycl.strftime("%Y") }} *</cycledef>
+  {% for dacycl_hr in dacycl_hrs %}
+  <cycledef group="{{dacycl_hr}}Z">{{ cdate_first_cycl.strftime("%Y%m%d") }}{{ dacycl_hr }}00 {{ cdate_last_cycl.strftime("%Y%m%d") }}{{ dacycl_hr }}00 24:00:00</cycledef>
+  {%- endfor %}
```

Comment:
I'm curious about the change to create multiple groups, one for each cycle time, versus giving a group of times a single label.

In general, a single label to a group of multiple forecast labels lets us see that these hours are DA cycles, and others are forecast cycles -- it doesn't necessarily matter from one XML to the next what those are, but the serve the same purpose.

Reply:
I agree it's a bit cumbersome when you look at the workflow.  But rocoto only allows two formats for the cycle groups -- start/end/step and cron-type format.  With either format, there are certain combinations of start/end times from config.sh that simply cannot be written out correctly.  Consider 20210123 and 20210201 ... with the cron format you'd get something like 23-01 for the day (wrong), and with the start/end/step, the DA cycles do not use a consistent step (e.g., 03, 06, 09, 15z, where the cold-start hours are skipped).  So my compromise was that you can at least look at the workflow and see which tasks are valid at which times.  This is very similar to the current RAP/HRRR workflows, which use multiple groups and names such as "12hr" for the 12z cycles

Reply:
List the cycle name as group name will give us a very flexible way to control the data assimilation cycles. We may further divide the da group to "catch up da"/"product da". But it also makes the xml task looks cumbersome. Think about we will have 15minutes cycles in the future. 

We do need more groups names to control the cycling, but most of the cycle's has the similar functions/dependences. I suggest we keep group name as less as we can. Right now, we only have two group "forecast" "dacycle". We may need to add one or two more groups names for cycling. But that should be how many groups we need from my current estimate.

We only have two ways to setup the cycle times in each group. In real-time, we can use * for month and day and only setup year. The most tricky part is how to setup cycle hour (and minute).

Reply:
OK well if you want to do that, we need to abandon the start/end times in the config file because there is no way to use them correctly.  Are we good with that, and willing to just use the cron format and <cycles> * * * <next few years> * ?

Reply:
Latest commit goes back to sensible group names for the cycle definitions, adds cycles for all days/months from starting year to [ending year]+1.  Also fixes a bug where ncl graphics were not zipped for da cycles